pkg/helmreconciler/reconciler.go

```diff
@@ -95,71 +100,53 @@ func (h *HelmReconciler) Reconcile() error {
 	// render charts
 	manifestMap, err := h.renderCharts(h.customizer.Input())
 	if err != nil {
-		err = errors.Wrap(err, fmt.Sprintf("error rendering charts"))
-		listenerErr := h.customizer.Listener().EndReconcile(h.instance, err)
-		if listenerErr != nil {
-			h.logger.Error(listenerErr, "unexpected error invoking EndReconcile")
-		}
 		return err
 	}
 
-	// determine processing order
-	chartOrder, err := h.customizer.Input().GetProcessingOrder(manifestMap)
-	if err != nil {
-		err = errors.Wrap(err, fmt.Sprintf("error ordering charts"))
-		listenerErr := h.customizer.Listener().EndReconcile(h.instance, err)
-		if listenerErr != nil {
-			h.logger.Error(listenerErr, "unexpected error invoking EndReconcile")
-		}
-		return err
-	}
-
-	// collect the errors.  from here on, we'll process everything with the assumption that any error is not fatal.
-	allErrors := []error{}
-
-	// process the charts
-	for _, chartName := range chartOrder {
-		chartManifests, ok := manifestMap[chartName]
-		if !ok {
-			// TODO: log warning about missing chart
-			continue
-		}
-		chartManifests, err := h.customizer.Listener().BeginChart(chartName, chartManifests)
-		if err != nil {
-			allErrors = append(allErrors, err)
-		}
-		err = h.ProcessManifests(chartManifests)
-		if err != nil {
-			allErrors = append(allErrors, err)
-		}
-		err = h.customizer.Listener().EndChart(chartName)
-		if err != nil {
-			allErrors = append(allErrors, err)
-		}
-	}
+	errs := h.processRecursive(manifestMap)
 
 	// delete any obsolete resources
```

Comment:
this comment could use some work.  I am left wondering what `obsolete` means in this context.

Reply:
Done.